### PR TITLE
Feature : Make Employee from Offer Letter

### PR DESCRIPTION
**Pull request by vjFaLk**
Also - 
Removed dead code from employee.js
Some formatting
Name, Email Id and Status are now mandatory for Job Application
Status field in Offer Letter is now no_copy

Fix for https://github.com/frappe/erpnext/issues/865

### CODE REVIEW

File: erpnext/hr/doctype/employee/employee.py [hunk 1]
```diff
@@ -167,19 +173,34 @@ def get_retirement_date(date_of_birth=None):
...
 	})
 	target.make_earn_ded_table()
 	return target
 
+
+@frappe.whitelist()
+def make_employee(source_name, target_doc=None):
+	def set_missing_values(source, target):
+		target.personal_email = frappe.db.get_value("Job Applicant", source.job_applicant, "email_id")
+	doclist = get_mapped_doc("Offer Letter", source_name, {
```
Comment (anandpdoshi) [inline erpnext/hr/doctype/employee/employee.py]: change `doclist` to `doc`

File: erpnext/hr/doctype/offer_letter/offer_letter.js [hunk 2]
```diff
@@ -6,5 +6,20 @@ frappe.ui.form.on("Offer Letter", {
 		frappe.model.get_value("Terms and Conditions", frm.doc.select_terms, "terms", function(value) {
 			frm.set_value("terms", value.terms);
 		});
+	},
+
+	refresh:function(frm){
+		if((!frm.doc.__islocal) && (frm.doc.status=='Accepted') && (frm.doc.docstatus===1)){
+		frm.add_custom_button(__('Make Employee'),
```
Comment (anandpdoshi) [inline erpnext/hr/doctype/offer_letter/offer_letter.js]: Indentation issue

File: erpnext/hr/doctype/offer_letter/offer_letter.js [hunk 3]
```diff
@@ -6,5 +6,20 @@ frappe.ui.form.on("Offer Letter", {
...
+		frm.cscript['Make Employee']);
 	}
+}
+});
+
+
+cur_frm.cscript['Make Employee'] = function() {
+frappe.model.open_mapped_doc({
```
Comment (anandpdoshi) [inline erpnext/hr/doctype/offer_letter/offer_letter.js]: indentation

File: erpnext/hr/doctype/offer_letter/offer_letter.js [hunk 4]
```diff
@@ -6,5 +6,20 @@ frappe.ui.form.on("Offer Letter", {
...
+	refresh:function(frm){
+		if((!frm.doc.__islocal) && (frm.doc.status=='Accepted') && (frm.doc.docstatus===1)){
+		frm.add_custom_button(__('Make Employee'),
+		frm.cscript['Make Employee']);
 	}
+}
+});
+
+
+cur_frm.cscript['Make Employee'] = function() {
```
Comment (anandpdoshi) [inline erpnext/hr/doctype/offer_letter/offer_letter.js]: Write this as:

```
frappe.provide("erpnext.offer_letter");
erpnext.offer_letter["Make Employee"] = function(frm) { ... }
```

File: erpnext/hr/doctype/offer_letter/offer_letter.js [hunk 5]
```diff
@@ -6,5 +6,20 @@ frappe.ui.form.on("Offer Letter", {
...
+frappe.model.open_mapped_doc({
+	method: "erpnext.hr.doctype.employee.employee.make_employee",
+	frm : cur_frm
```
Comment (anandpdoshi) [inline erpnext/hr/doctype/offer_letter/offer_letter.js]: avoid `cur_frm` as much as possible

**Comment by vjFaLk** [comment]
Made changes accordingly.

**Comment by anandpdoshi** [comment]
merged manually

**Pull request closed without merging**